ntoskrnl/ex/handle.c

```diff
@@ -728,16 +728,11 @@ ExpAllocateHandleTableEntry(IN PHANDLE_TABLE HandleTable,
             KeLeaveCriticalRegion();
             OldValue = HandleTable->FirstFree;
 
-            /* Check if allocation failed */
-            if (!Result)
-            {
-                /* Check if nobody else went through here */
-                if (!OldValue)
-                {
-                    /* We're still the only thread around, so fail */
-                    NewHandle->GenericHandleOverlay = NULL;
-                    return NULL;
-                }
+            /* Check if allocation failed and nobody else went through here */
+            if (!Result && !OldValue)
+                /* We're still the only thread around, so fail */
+                NewHandle->GenericHandleOverlay = NULL;
+                return NULL;
```

Comment:
You probably introduced a bug here.

Reply:
{ }

Reply:
Sure. I re-added the `{` line which was unintentionaly removed.